src/data.rs

```diff
@@ -169,6 +172,66 @@ impl<TYPES: NodeType> VidDisperse<TYPES> {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, Hash)]
+pub struct VidDisperseShare<TYPES: NodeType> {
+    /// The view number for which this VID data is intended
+    pub view_number: TYPES::Time,
+    /// Block payload commitment
+    pub payload_commitment: VidCommitment,
+    /// A storage node's key and its corresponding VID share
+    pub share: VidShare,
+    /// VID common data sent to all storage nodes
+    pub common: VidCommon,
+    /// a public key of the share recipient
+    pub recipient_key: TYPES::SignatureKey,
+}
+
+impl<TYPES: NodeType> VidDisperseShare<TYPES> {
+    /// Create a vector of `VidDisperseShare` from `VidDisperse`
+    pub fn from_vid_disperse(
+        vid_disperse: VidDisperse<TYPES>,
+    ) -> Vec<Self> {
+        vid_disperse.shares.into_iter().map(|(recipient_key, share)| {
+            VidDisperseShare {
+                share,
+                recipient_key,
+                view_number: vid_disperse.view_number,
+                common: vid_disperse.common.clone(),
+                payload_commitment: vid_disperse.payload_commitment,
+            }
+        }).collect()
+    }
+
+    /// Create a vector of `Proposal`s of `VidDisperseShare`
+    pub fn from_vid_disperse_proposal(
```

Comment:
Do you have a use for this one in mind?  It seems like the leader would use the function above and below this to transform the VidDisperse it calculates into a Vec of shares then call to proposal on the shares?  Either way the idea of these helpers seems really good.